app/events/base_event_store_event.rb

```diff
@@ -10,6 +10,10 @@ module Categorizable
     end
   end
 
+  def initialize(args = {})
```

Comment:
Support for Ruby 3.1 is added in `rails_event_store` version `v2.4.0`: https://github.com/RailsEventStore/rails_event_store/releases/tag/v2.4.0

But we are currently at `0.9.0`... :melting_face:

Reply:
I am not sure, but it looks like a very big jump :upside_down_face: 
Probably would involve some DB migrations. So maybe we can attempt to upgrade it (I think that would be nice), but I'd suggest to do it in another PR :)

Reply:
Oh, I didn't know that was so demanding. Surely if we do it it must be in another PR.

Reply:
Yeah, at least these releases have some breaking changes and required DB migrations:
https://github.com/RailsEventStore/rails_event_store/releases/tag/v0.19.0
https://github.com/RailsEventStore/rails_event_store/releases/tag/v2.0.0

I'm not sure it's even possible to upgrade in one go... needs to be investigated.

Reply:
I read this upgrade walkthrough [1] in the past and decided that upgrading it would be probably harder than replacing it. I don't think we want to have this event system. Hard to track it's logic and we have so many other approaches already.

But if anybody wants to...

[1] https://blog.arkency.com/update-rails-event-store-to-v1-dot-0-0-walkthrough/

Reply:
I don't mean to start a full-blown discussion on what to do with the event store, but...

> Hard to track it's logic and we have so many other approaches already.

@akostadinov which approaches you are referring to?

Reply:
I mean that we use many design patterns. hooks/callbacks, services, events, background jobs, and maybe more partially competing design patterns. IMO services is easier to maintain than events. FWIW on last f2f Hery expressed an opinion that in the long term services are easier to understand and maintain in the long run. That's what I meant. But yeah, open for discussion.